pyomo/core/base/constraint.py

```diff
@@ -64,6 +64,7 @@
     InequalityExpression,
     RangedExpression,
 }
+_strict_relational_exprs = {True, (False, True), (True, False), (True, True)}
```

Comment:
Do you really need to declare this as a global var if you're only using it once?

Reply:
Yes: It is used once, but for every ConstraintData... so for a 200M constraint problem, that would create and throw away the set 200M times 

[This is one of the weird(er) performance tricks: if you are looking up into a group of more than ~2 things, it is faster to use a set than, say, a list - but only if you are not creating & throwing the set away each time]